File: desert/lurkers.lua
function attack(e)
    if e.health > 33 and self:isReady("bash") then
        self:bash(e)
    else
        self:attack(e)
    end
end

local enemies = self:findEnemies()
local enemyIndex = 1

loop
    local enemy = enemies[enemyIndex]
    if enemy then
        if enemy.type == 'shaman' then
            while enemy.health > 0 do
                attack(enemy)
            end
        end
    end
    enemyIndex = enemyIndex + 1
end

File: desert/lurkers.clj
(defn attack [e]
  (if (.isReady this "power-up")
    (when true
      (.powerUp this)
      (.attack this e))
    (if (.isReady this "bash")
      (.bash this e)
      (.attack this e))))

(while true
  (def es (.findEnemies this))
  (doseq [e es]
    (if (= (.type e) "shaman")
      (while (> (.health e) 0)
        (attack e))
      (.wait this 0.1))))
